src/stores/GuildEmojiRoleStore.js

```diff
@@ -82,12 +90,15 @@ class GuildEmojiRoleStore extends DataStore {
    * @private
    */
   _patch(roles) {
-    this.clear();
+    this.emoji._roles = roles;
+  }
 
-    for (let role of roles) {
-      role = this.guild.roles.resolve(role);
-      if (role) super.set(role.id, role);
-    }
+  *[Symbol.iterator]() {
+    yield* this._filtered.entries();
+  }
+
+  valueOf() {
+    return this._filtered;
   }
 
   /**
```

Comment:
Are the docstrings here still applicable now? (You removed equivalent docstrings in the `GuildMemberRoleStore`)